src/doom/g_game.c

```diff
@@ -683,18 +683,42 @@ void G_BuildTiccmd (ticcmd_t* cmd, int maketic)
     {
         static unsigned int kbdlookctrl = 0;
 
-        if (gamekeydown[key_lookup] || joylook < 0)
+        if (gamekeydown[key_lookup])
         {
             look = lspeed;
             kbdlookctrl += ticdup;
         }
         else
-        if (gamekeydown[key_lookdown] || joylook > 0)
+        if (gamekeydown[key_lookdown])
         {
             look = -lspeed;
             kbdlookctrl += ticdup;
         }
         else
+        if (use_analog && joylook)
+        {
+            joylook = joylook * joystick_look_sensitivity / 10;
+            joylook = (joylook > FRACUNIT) ? FRACUNIT : joylook;
+            joylook = (joylook < -FRACUNIT) ? -FRACUNIT : joylook;
+            look = -FixedMul(2, joylook);
+            kbdlookctrl += ticdup;
+        }
+        else
+        if (joystick_look_sensitivity && joylook)
```

Comment:
I don't think I full fully understand this sequence of conditions. Could you add a commit or two, please?

Reply:
The `&& joylook` was a mistake and I've removed it.

The intent of checking for non-zero `joystick_*_sensitivity` is to allow a sensitivity setting of 0 to disable that joystick axis regardless of the `use_analog` setting. See: [7bd0290](https://github.com/chocolate-doom/chocolate-doom/pull/1629/commits/7bd029058c7be03cff6a151925c1c424a4651d8b)

Reply:
Does 1ab437f help deobfuscate things? (If so, should I apply similar refactoring to the other Chocolate joystick code? It probably makes sense to do that upstream rather than in Crispy.)